### DISCUSSION TITLE
regex_syntax::Parser.parse and regex::Regex.build alignment?

### DISCUSSION THREAD

**Discussion by Guillermogsjc** [rust-lang/regex#937] [Q&A] [answered]
Hi,

is it supposed that  `regex_syntax::Parser`  method `parse` to return `Err` in the same cases that `regex:Regex` method `build` ?

My tries tell me that this assumption is correct, but do not know if it is and will be a criteria in this repository and it will be aligned in the future.

Regards

**Comment by BurntSushi** [comment, marked as answer]
Sorry, it is very difficult to understand your question.

I think what you're asking is whether Regex::new can fail due to anything except a parse error. Indeed it can! It can fail if the regex exceeds the configured RegexBuilder::size_limit.

**Reply by Guillermogsjc** [reply]
Sorry for my bad questioning, wanted to ask if:

 regex_syntax::Parser.parse rising Err for a pattern, implies it will fail to build with regex::Regex for the same pattern?

**Reply by BurntSushi** [reply]
Ah. If the configuration is the same for both, then yes, that's correct. If it isn't so, please file a bug.
